examples/run/run.cpp

```diff
@@ -1053,11 +1074,43 @@ static int get_user_input(std::string & user_input, const std::string & user) {
     return 0;
 }
 
+// Reads a chat template file to be used
+static std::string read_chat_template_file(const std::string & chat_template_file) {
+    if(chat_template_file.empty()){
+        return "";
+    }
+
+    FILE* file = ggml_fopen(chat_template_file.c_str(), "r");
```

Comment:
This would be better if it used the File class from this code. It would automatically fclose the file when necessary. I would also change fopen -> ggml_fopen in that File class.

Reply:
The file class from this code is behind the `LLAMA_USE_CURL` macro. So I'd need to move it since the `--chat-template` should also work without that macro. What do you think?

Reply:
Yeah I'd move it out of the LLAMA_USE_CURL macro. It doesn't actually do any curl stuff.